core/src/main/java/io/kestra/core/test/flow/Assertion.java

```diff
@@ -22,14 +26,234 @@ public class Assertion {
     private Property<String> endsWith;
     private Property<String> startsWith;
     private Property<String> contains;
-    private Property<String> equalsTo;
-    private Property<String> notEqualsTo;
+    private Property<Object> equalsTo;
+    private Property<Object> notEqualsTo;
     private Property<Double> greaterThan;
     private Property<Double> greaterThanOrEqualTo;
     private Property<Double> lessThan;
-    private Property<Double> lesThanOrEqualTo;
+    private Property<Double> lessThanOrEqualTo;
     private Property<List<String>> in;
     private Property<List<String>> notIn;
     private Property<Boolean> isNull;
     private Property<Boolean> isNotNull;
+
+    public List<AssertionResult> run(RunContext runContext) {
+        try {
+            var actualValue = runContext.render(this.getValue()).as(Object.class).orElse(null);
+            var errorMessage = runContext.render(this.getErrorMessage()).as(String.class);
+            var description = runContext.render(this.getDescription()).as(String.class);
+            var taskId = Optional.ofNullable(this.getTaskId());
+
+            var results = new ArrayList<AssertionResult>();
+
+            var isNull = runContext.render(this.getIsNull()).as(Boolean.class);
+            var isNotNull = runContext.render(this.getIsNotNull()).as(Boolean.class);
+            if (isNull.isPresent()) {
+                return List.of(new AssertionResult(
+                    "isNull",
+                    isNull.get(),
+                    actualValue,
+                    isNull(isNull.get(), actualValue),
+                    taskId.orElse(null),
+                    description.orElse(null),
+                    errorMessage.orElse(null)
```

Comment:
Please provide a good error message if not specified by the user.
For ex: 'expected `actualValue` to be null but is `{{actualValue}}`.
The same for all error message, the first is literally what the user defined in the assertion, the second is the rendered value.